public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,833 @@
+  }
+}
+
+export class FilterDisplay extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    let filter = this.props;
+    if (filter === 'all fields are included') {
+      return (
+        <EuiText>
+          <p className="enabled">-</p>
+        </EuiText>
+      );
+    } else {
+      return <EuiText>{filter}</EuiText>;
+    }
+  }
+}
+const FixedWidthRow = (props) => <EuiFormRow {...props} style={{ width: '150px' }} />;
+
+async function createAndStartDetector(context) {
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors', {
+      configs,
+    });
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    const detectorId = _id;
+    if (ok) {
+      try {
+        const response = await httpClient.post(`../api/alerting/detectors/${detectorId}/_start`);
+        const {
+          data: {
+            ok,
+            response: { _id },
+          },
+        } = response;
+        if (ok) {
+          context.setFlyout(null);
+          context.renderStartedDetectorFlyout(configs, _id, context.queriesForOverview);
+        }
+      } catch (err) {
+        if (typeof err === 'string') throw err;
+        throw 'There was a problem starting detector';
+      }
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
+    throw 'There was a problem createing detector';
+  }
+}
+
+function warningOrSubduedallOut(message, color, iconType) {
+  return (
+    <EuiCallOut color={color}>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiIcon type={iconType} />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          {' '}
+          <EuiTextColor>{message}</EuiTextColor>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiCallOut>
+  );
+}
+
+function triggerCorrectCallOut(context) {
+  if (context.startedDetector) {
+    return informationCallOut(context, calloutType);
+  }
+  let calloutType = 'notValid';
+  if (
+    Object.keys(context.failures).length == 0 &&
+    Object.keys(context.suggestedChanges).length == 0
+  ) {
+    calloutType = 'valid';
+  }
+  for (let [key, value] of Object.entries(context.suggestedChanges)) {
+    if (key === 'detection_interval') {
+      let intervalMinutes = Math.ceil(value[0] / 60000) + 1;
+      if (isNaN(intervalMinutes) || intervalMinutes > MAX_INTERVAl_LENGTH_MINUTES) {
+        calloutType = 'maxInterval';
+        context.adConfigs.detection_interval = {
+          period: { interval: MAX_INTERVAl_LENGTH_MINUTES, unit: 'MINUTES' },
+        };
+      } else {
+        context.adConfigs.detection_interval = {
+          period: { interval: intervalMinutes, unit: 'MINUTES' },
+        };
+        if (
+          Object.keys(context.failures).length == 0 &&
+          Object.keys(context.suggestedChanges).length == 1
+        ) {
+          calloutType = 'valid';
+        }
```

Comment:
This should be ran outside of the `for` loop, and if true, never execute the `for` loop. Also, why would there be any suggested changes if the callout is valid?

Reply:
This is used for the case where the suggested change is a new detector interval (a specific number and not the message that the max was reached) so at that case I just update the configurations and the callout is hence valid. I'll try to make this logic cleaner in my code

Reply:
Got it, yeah. Sounds like a specific case that could be outside of this loop